File: src/test/java/de/pflugradts/passbird/domain/model/namespace/NamespaceTest.java
package de.pflugradts.passbird.domain.model.namespace;

import de.pflugradts.passbird.domain.model.transfer.Bytes;
import org.junit.jupiter.api.Test;

import static de.pflugradts.passbird.domain.model.namespace.NamespaceSlot.DEFAULT;
import static org.assertj.core.api.Assertions.assertThat;

public class NamespaceTest {

    @Test
    void shouldCreateNamespace() {
        // given / when / then
        assertThat(Namespace.create(Bytes.bytesOf("namespace"), DEFAULT)).isNotNull();
    }

    @Test
    void shouldCloneBytes() {
        // given
        final var bytes = Bytes.bytesOf("key");
        final var namespace = Namespace.create(bytes, DEFAULT);

        // when
        bytes.scramble();
        final var actual = namespace.getBytes();

        // then
        assertThat(actual).isNotNull().isNotEqualTo(bytes);
    }

    @Test
    void shouldCreateDefaultNamespace() {
        // given / when
        final var defaultNamespace = Namespace.DEFAULT;

        // then
        assertThat(defaultNamespace).isNotNull();
        assertThat(defaultNamespace.getBytes()).isNotNull().isEqualTo(Bytes.bytesOf("Default"));
        assertThat(defaultNamespace.getSlot()).isNotNull().isEqualTo(DEFAULT);
    }

}

File: src/test/kotlin/de/pflugradts/passbird/domain/model/namespace/NamespaceTest.kt
package de.pflugradts.passbird.domain.model.namespace

import de.pflugradts.passbird.domain.model.namespace.Namespace.Companion.DEFAULT
import de.pflugradts.passbird.domain.model.namespace.Namespace.Companion.createNamespace
import de.pflugradts.passbird.domain.model.transfer.Bytes.Companion.bytesOf
import org.junit.jupiter.api.Test
import strikt.api.expectThat
import strikt.assertions.isEqualTo
import strikt.assertions.isNotEqualTo

class NamespaceTest {
    @Test
    fun `should create namespace`() {
        // given
        val name = "namespace"

        // when
        val actual = createNamespace(bytesOf("namespace"), NamespaceSlot.DEFAULT)

        // then
        expectThat(actual.bytes.asString()) isEqualTo name
    }

    @Test
    fun `should clone bytes`() {
        // given
        val bytes = bytesOf("key")
        val namespace = createNamespace(bytes, NamespaceSlot.DEFAULT)

        // when
        bytes.scramble()
        val actual = namespace.bytes

        // then
        expectThat(actual) isNotEqualTo bytes
    }

    @Test
    fun `should create default namespace`() {
        // given / when
        val defaultNamespace = DEFAULT

        // then
        expectThat(defaultNamespace.bytes) isEqualTo bytesOf("Default")
        expectThat(defaultNamespace.slot) isEqualTo NamespaceSlot.DEFAULT
    }
}
